internal/handler/response.go

```diff
@@ -14,4 +15,7 @@ func writeResponse(w http.ResponseWriter, code int, v interface{}) {
 	}
 	w.WriteHeader(code)
 	w.Write([]byte(b))
+	if err != nil {
```

Comment:
What error are you checking?
Did you mean `if _, err = w.Write(...); err != nil {...}`?